cmake/pybind11.cmake

```diff
@@ -0,0 +1,29 @@
+#
+# Copyright (c) 2019-2020, New York University and Max Planck Gesellschaft.
+# License BSD-3 clause
+#
+
+#.rst:
+#
+# Add a Python module using pybind11
+#
+# .. cmake:command:: ADD_PYBIND11_MODULE
+#
+#    Add a pybind11 module.  Takes as argument the name of the module followed
+#    by an optional list of libraries which are linked to the target.
+#
+#    This macro expects that the code is defined in a single source file
+#    "srcpy/${module_name}.cpp"!
+#
+macro(add_pybind11_module module_name)
```

Comment:
we should fix the arguments in this PR.

Reply:
I can help on this if I have time...

Reply:
Okay, I'll look into it tomorrow.  What should the options be? My suggestions would be:
- `LINK_LIBRARIES` --> stuff that goes to `target_link_libraries`
- `INCLUDE_DIRS` --> stuff that goes to `target_include_directories`.  Should "include" be there by default? This might not be needed anymore as include directories are now provided by link targets anyway.
- `SOURCE` --> source files that go to `pybind11_add_module`.  Should we keep the default of `srcpy/${module_name}.cpp`?  It's convenient but maybe a bit dirty.

Reply:
ok for `LINK_LIBRARIES`, this could be targets or libraries anyway.
`INCLUDE_DIRS` should definitely be optional with the "include" dir by default.
`SOURCE` is good for the source files.
We should not keep the `srcpy/${module_name}.cpp` it would make things dirtyer later on I guess...

Reply:
Done.  I did not use a name for the source files, though, to be more in line with standard commands like `add_library`.  It looks now like this:

```cmake
add_pybind11_module(target_name srcpy/file1.cpp, srcpy/file2.cpp
    INCLUDE_DIRS  # "include" is added automatically
       ${OpenCV_INLCUDE_DIRS}
    LINK_LIBRARIES
       lib1
       lib2
)
```

Reply:
Perfect thank you very much!!!